rock/orogen_plugin.rb

```diff
@@ -727,6 +727,32 @@ def update_spec
                         doc("the global name that should be used for the internal #{name} frame")
                 end
             end
+            
+            #also do frame mapping in case the transformations are requested
+            frame_selection = configurable_frames.sort.map do |frame_name|
+                "    _#{name}.setFrameMapping(\"#{frame_name}\", _#{frame_name}_frame);"
+            end.compact
+            #add method to request needed transformations
+            transformationBody = frame_selection.join("\n") + " \n
+    std::vector<base::samples::RigidBodyState > ret;
+    const std::vector<transformer::Transformation *> &transformations(_transformer.getRegisteredTransformations());
+    ret.reserve(transformations.size());
+    
+    for(std::vector<transformer::Transformation *>::const_iterator transform = transformations.begin();
+        transform != transformations.end(); transform++)
+    {
+        base::samples::RigidBodyState curTr;
+        curTr.sourceFrame = (*transform)->getSourceFrame();
+        curTr.targetFrame = (*transform)->getTargetFrame();
+        
+        ret.push_back(curTr);
+    }
+    
+    return ret;"
+
+            task.hidden_operation("getNeededTransformations", transformationBody).
```

Comment:
This is borken i just created a patch (*untested*) https://github.com/goldhoorn/drivers-orogen-transformer/commit/5f1e3725b4fad1f7899a4605d7d0c6704de98dc7  which fixes the hidden operation but it depends on https://github.com/orocos-toolchain/orogen/pull/40